contracts/zero-ex/contracts/src/features/UniswapFeature.sol

```diff
@@ -389,6 +360,103 @@ contract UniswapFeature is
                 returndatacopy(0, 0, returndatasize())
                 revert(0, returndatasize())
             }
+
+            // Move `amount` tokens from the taker/caller to `to`.
+            function moveTakerTokensTo(token, to, amount) {
+
+                // If the token is possibly greedy, we check the allowance rather
+                // than relying on letting the transferFrom() call fail and
+                // falling through to legacy allowance target because the token
+                // will eat all our gas.
+                if isTokenPossiblyGreedy(token) {
+                    // Check if we have enough direct allowance.
+                    mstore(0xB00, ALLOWANCE_CALL_SELECTOR_32)
+                    mstore(0xB04, caller())
+                    mstore(0xB24, address())
+                    if not(iszero(call(gas(), token, 0, 0xB00, 0x44, 0xC00, 0x20))) {
```

Comment:
Oh man, I wonder if there are calls to `allowance` that could `assert`. Theoretically this should just be a mapping that we're querying, so same as the other note.. Just depends how defensive we want to be here.

Reply:
As well, should we revert if this call fails for some reason? 🤔

Reply:
I think a token that throws during an allowance call blows up other of the stack anyway.